supernode/services/userdataprocess/service.go

```diff
@@ -43,7 +46,13 @@ func (service *Service) Run(ctx context.Context) error {
 
 // Task returns the task of the registration userdata by the given id.
 func (service *Service) Task(id string) *Task {
-	return service.Worker.Task(id).(*Task)
+	for i := 0; i < getTaskRetry; i++ {
+		if task := service.Worker.Task(id); task != nil {
+			return task.(*Task)
+		}
+		time.Sleep(getTaskRetryInterval)
+	}
```

Comment:
nit: could you add this logic inside worker.Task()?? I saw this is happen several times in our code